rust/src/writer/json.rs

```diff
@@ -169,11 +166,11 @@ impl DataArrowWriter {
     }
 
     fn new_underlying_writer(
-        cursor: InMemoryWriteableCursor,
+        buffer: ShareableBuffer,
         arrow_schema: Arc<ArrowSchema>,
         writer_properties: WriterProperties,
-    ) -> Result<ArrowWriter<InMemoryWriteableCursor>, ParquetError> {
-        ArrowWriter::try_new(cursor, arrow_schema, Some(writer_properties))
+    ) -> Result<ArrowWriter<ShareableBuffer>, ParquetError> {
+        ArrowWriter::try_new(buffer, arrow_schema, Some(writer_properties))
     }
 }
```

Comment:
i do remember stumbling across this before and not introduced in this PR, but wrapping this one-liner in a function does not seem to add convenience :).

Reply:
I almost changed this and a few other things in this PR, but I decided to save it for the followup. I agree it isn't useful in its current state, but I'd still like to address it next time.